cli: web-components stories homogenization

tooppaaa opened the pull request:
Issue:

We need similar stories after generation to run e2e against

Part of #10702

tooppaaa committed:
cli: web-components stories homogenization

gaetanmaisse commented on lib/cli/src/generators/WEB-COMPONENTS/template/stories/0-Welcome.stories.js:
```diff
@@ -0,0 +1,49 @@
+    </a>
+    section in our documentation.
+  </p>
+  <p class="note">
+    <b>NOTE:</b> <br />
+    Have a look at the <span class="inline-code">.storybook/webpack.config.js</span> to add webpack
+    loaders and plugins you are using in this project.
+  </p>
```
I think we can remove that as I'm not sure there will be a `.storybook/webpack.config.js`

yannbf replied:
You are right, the template does not contain such file:
![image](https://user-images.githubusercontent.com/1671563/81496423-e365ee00-92b7-11ea-98eb-239c300ebd7f.png)

However it's nice to tell people that if they add a `webpack.config.js` file it will be picked up, so maybe just rephrase that? Also, that sentence is in every framework's template, so if we were to make a change, it has to be synced with the others.

@shilman what's your take on this?

shilman replied:
AFAIK the current recommendation is to override `webpackFinal` in `main.js`, so we should probably update all the docs accordingly. `webpack.config.js` is still supported (for now, not sure how long we plan to keep this? But we should be  directing everybody towards the recommendation. @ndelangen please feel free to correct if I got any of this wrong.

ndelangen replied:
> the current recommendation is to override webpackFinal in main.js

indeed

gaetanmaisse commented on lib/cli/src/generators/WEB-COMPONENTS/index.js:
```diff
@@ -15,17 +14,18 @@ export default async (npmOptions, { storyFormat }) => {
     '@storybook/web-components',
     '@storybook/addon-actions',
     '@storybook/addon-links',
+    '@storybook/addons',
```
Is `@storybook/addons` really needed directly as it's already a dep of `@storybook/addon-actions` and `@storybook/addon-links`? Maybe @ndelangen have some info about that

ndelangen replied:
If it works without, let's remove it!

yannbf replied:
@tooppaaa the other points have been resolved and will be done in a different PR, but can you check this one? then we're good to merge!

gaetanmaisse replied:
https://github.com/storybookjs/storybook/pull/10703#issuecomment-628227892

yannbf commented on lib/cli/src/generators/WEB-COMPONENTS/template/stories/1-Button.stories.js:
```diff
@@ -0,0 +1,42 @@
+`;
+
+Emoji.story = {
+  parameters: { notes: 'My notes on a button with emojis' },
```
I think we should get rid of `notes` for every framework. It's deprecated or was even removed already.
@ndelangen what do you think?

gaetanmaisse replied:
You're right @yannbf, `notes` has been moved to the repo `storybookjs/deprecated-addons` so no 6.0 version will be released.

shilman replied:
Agree we should get rid of `notes` and pick a better example. Maybe `backgrounds`?

ndelangen replied:
I agree, we should leave it out for now.

yannbf reviewed (changes requested):
@tooppaaa  the other points have been resolved and will be done in a different PR, but can you check the one about removing `@storybook/addons`? then we're good to merge!

gaetanmaisse committed:
feat(cli): remove `@storybook/addons` from deps installed when running `sb init` in a web-component project

`@storybook/addons` is already a dependency of all `@storybook/addon-XXX` packages so there is no need to add it to the root project directly.

gaetanmaisse committed:
Merge branch 'next' into generator_webComponentsStories

gaetanmaisse commented:
Tested ✅ `@storybook/addons` is already a dependency of all `@storybook/addon-XXX` packages so there is no need to add it to the root project directly.
